### PR TITLE
FIX: Update `optimization_tutorial.py` to expose `batch_size` in `train_loop()`

### PR DESCRIPTION
**Pull request by loganthomas**
## Description
`batch_size` is not currently exposed as `train_loop()` parameter
- Re-implements #2920 (accidental deletion from forked repo)

## Checklist

- [X] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [X] Only one issue is addressed in this pull request
- [X] Labels from the issue that this PR is fixing are added to this pull request
- [X] No unnecessary issues are included into this pull request.

cc @subramen @albanD @sekyondaMeta @svekars @kit1980 @brycebortree

### CODE REVIEW

**Comment by subramen** [comment]
Hi, thanks for the PR but the proposed change is not correct. The `dataloader` takes in the batch size arg (as seen here https://github.com/pytorch/tutorials/blob/0740801fa8df2464f4d931d36a4bd7ada9cd0519/beginner_source/basics/optimization_tutorial.py#L47)

**Comment by loganthomas** [comment]
> The `dataloader` takes in the batch size arg (as seen here)


Yes, but part of the `train_loop` uses the `batch_size` for reporting: https://github.com/pytorch/tutorials/blob/0740801fa8df2464f4d931d36a4bd7ada9cd0519/beginner_source/basics/optimization_tutorial.py#L166
